keanu-project/src/test/java/io/improbable/keanu/vertices/AssertVertexTest.java

```diff
@@ -0,0 +1,88 @@
+package io.improbable.keanu.vertices;
+
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.algorithms.variational.optimizer.KeanuOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.ConstantBoolVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class AssertVertexTest {
```

Comment:
Nice tests! Would be good to see one where we check it handles value propagation correctly. Assert that a vertex is less than a value, expect it to pass, change the vertex's value to now be much larger, assert that it fails.

Also - we use the optimiser / sampler and RV's quite a bit in this test class, can you set the `DeterministicRule`?

Reply:
Good idea. Have added two which I think should cover this now. They check whether both a valid and invalid `setAndCascade` has the correct behaviour. Also added `DeterministicRule`.